core/src/main/scala/org/apache/spark/util/random/RandomSampler.scala

```diff
@@ -41,6 +41,12 @@ trait RandomSampler[T, U] extends Pseudorandom with Cloneable with Serializable
   /** take a random sample */
   def sample(items: Iterator[T]): Iterator[U]
```

Comment:
Would it make sense to either add a default method to this so that individuals implementing their own sampler don't have to duplicate the sample logic or just deprecate or remove this method if we don't need it anymore?